src/core/server/core_app/core_app.ts

```diff
@@ -49,4 +53,12 @@ export class CoreApp {
       res.ok({ body: { version: '0.0.1' } })
     );
   }
+  private registerStaticDirs(coreSetup: InternalCoreSetup) {
+    coreSetup.http.registerStaticDir('/ui/{path*}', Path.resolve(__dirname, './assets'));
```

Comment:
I'm open to the other options. we can place them under `src/core/server/assets`, for example.

Reply:
I know this PR doesn't yet expose this to plugins directly, but I wonder if there's really any value into allowing/requiring plugins to add the `{path*}` part of the path string. Should we do this for them automatically?

I think that if there is a real use case to only serving files of a certain directory depth, this should be exposed more explicitly via an option to `registerStaticDir`:

```
registerStaticDir({
  path: '/ui/',
  directory: Path.resolve(__dirname, './assets',
  depth: 3 // accepts positive integers, defaults to `Infinity`
})
```

Can be dealt with in PR that exposes to plugins. Just wanted to note while I thought about it. Also if there's no current need to specify the depth we probably don't need this option at all and should just automatically append the `/{path*}` when we register the route with Hapi.

Reply:
> Can be dealt with in PR that exposes to plugins.

I don't think we should expose this API directly to the plugins. Otherwise, any plugin can expose an arbitrary folder.

Reply:
We expose `/public` directory in the LP automatically: https://github.com/elastic/kibana/blob/85c8232c0b81e9e0dd2265e04b2cc26dffaf1b60/src/legacy/plugin_discovery/plugin_spec/plugin_spec.js#L164-L174

Reply:
> I don't think we should expose this API directly to the plugins. Otherwise, any plugin can expose an arbitrary folder (config folder with kibana.yml, for example)

We could have an API to only allow to serve from within the plugin's base folder. That would avoid the risk of exposing files from outside their folder.

i.e

```ts
// src/core/plugins/my_plugin/server/plugin.ts
coreSetup.http.registerStaticDir('/my-plugin-data/{path*}', 'my_assets_folder');
```

that would serve the `src/core/plugins/my_plugin/my_assets_folder` (or maybe `src/core/plugins/my_plugin/server/my_assets_folder`, need to decide)

> I know this PR doesn't yet expose this to plugins directly, but I wonder if there's really any value into allowing/requiring plugins to add the {path*}

I agree, that doesn't seems necessary imho

Previous proposal example should probably be

```ts
coreSetup.http.registerStaticDir('/my-plugin-data', 'my_assets_folder');
```

> I think that if there is a real use case to only serving files of a certain directory depth, this should be exposed more explicitly via an option to registerStaticDir

Don't really see any obvious need for that. KISS imho.

Reply:
If we don't need to expose to plugins, then as-is is good with me 👍